ext/string_metric/levenshtein/iterative_with_two_matrix_rows_ext.c

```diff
@@ -0,0 +1,99 @@
+#include <ruby.h>
+
+#define MIN(a, b, c) ((a) < (b) ? ((a) < (c) ? (a) : (c)) : ((b) < (c) ? (b) : (c)))
+
+#define MALLOC_W(ptr, size) \
```

Comment:
Multiline macros should always be enclosed in a `do while (0)` (or similar) to properly expand it into a statement.
Not using them can hide some very subtle bugs. Let's say i call this macro inside an if-else statement:

    if (condition)
        MALLOC_W(ptr, 5);
    else
        return;

and here it is after the preprocessor will expand it:

    if (condition)
      if ((ptr = malloc(5)) == ((void *)0)) rb_memerror();
    else
      return;

Notice that the MALLOC_W macro "eats" the else clause from the first `if`.

Properly enclosed in a do-while, yields the correct result:

    if (condition)
      do { if ((ptr = malloc(5)) == ((void *)0)) rb_memerror() } while (0);
    else
      return;

For more information: 
http://stackoverflow.com/questions/1067226/c-multi-line-macro-do-while0-vs-scope-block
http://stackoverflow.com/questions/154136/do-while-and-if-else-statements-in-c-c-macros